Dockerfile

```diff
@@ -1,39 +1,53 @@
-FROM golang AS aws-k8s-tester-builder
+ARG GO_VERSION
+ARG AL_VERSION
+
+FROM golang:${GO_VERSION} AS aws-k8s-tester-builder
+ARG GO_PROXY
 RUN go version
 ADD ./ /go/src/github.com/aws/aws-k8s-tester
 WORKDIR /go/src/github.com/aws/aws-k8s-tester
-ARG RELEASE_VERSION=latest
-ARG OS_TARGETS=linux
-RUN ./hack/build.sh
+RUN GO_PROXY=${GO_PROXY} ./hack/build.sh
+
+FROM golang:${GO_VERSION} AS clusterloader2-builder
+ARG GO_PROXY
```

Comment:
The `ARG` is an environment variable, so if you call it `GOPROXY` here, you don't need to convert it in every `RUN` (i.e. with `GOPROXY=${GO_PROXY}`).

Similarly, if you pass `--build-arg GOPROXY=${GOPROXY}` in the `Makefile`, you shouldn't need the changes in `hack/build.sh` --`go` will pick up the value from the env.